mlflow/xgboost.py

```diff
@@ -382,7 +382,11 @@ def record_eval_results(eval_results, metrics_logger):
             Create a callback function that records evaluation results.
             """
 
-            if LooseVersion(xgboost.__version__) >= LooseVersion("1.3.0"):
+            # The dev version of XGBoost (e.g. '1.5.0-SNAPSHOT') doesn't match the scheme
+            # defined in PEP 440 and causes `packaging.version.Version` to fail.
+            # As a workaround, replace 'SNAPSHOT' with 'dev':
+            # https://github.com/dmlc/xgboost/blob/81bdfb835d0d0244c86649e74d2dc3c00beb3942/python-package/xgboost/VERSION#L1  # noqa
+            if Version(xgboost.__version__.replace("SNAPSHOT", "dev")) >= Version("1.3.0"):
```

Comment:
XGBoost needs some special handling.

Reply:
Filed a proposal here: https://github.com/dmlc/xgboost/issues/6984

Reply:
@WeichenXu123 One of the core maintainer in XGBoost created a PR for the issue above:

https://github.com/dmlc/xgboost/pull/6988